kobo/log.py

```diff
@@ -76,21 +77,25 @@ def add_file_logger(logger, logfile, log_level=None, format=None, mode="a"):
     log_level = log_level or logging.DEBUG
     format = format or BRIEF_LOG_FORMAT
 
+    # Create parent directory if needed.
+    try:
+        os.makedirs(os.path.dirname(logfile))
+    except OSError as exc:
+        if exc.errno != errno.EEXIST:
+            raise
+
     # touch the logfile
     if not os.path.exists(logfile):
-        try:
-            fo = open(logfile, "w")
-            fo.close()
-        except (ValueError, IOError):
-            return
+        fo = open(logfile, "w")
+        fo.close()
 
     # is the logfile really a file?
     if not os.path.isfile(logfile):
```

Comment:
I'm nervous about this change - since the behavior previously was "silently do nothing" on errors, there could be systems depending on that now.

That's true of all the conditionals but I see this one as the most problematic because os.path.isfile returns false on /dev/null.

So, consider the scenario that some tool uses add_file_logger with the destination read from a config file.  User could previously set the destination to /dev/null if they didn't care about this log.  Now that will crash.

I'd like to suggest following changes for consideration:

- Just drop the isfile check.  I can't see the point.  FileHandler already raises a reasonable error if you try to use it on a non-file, e.g. "IOError: [Errno 21] Is a directory: '/tmp'", and works normally if you give it a writable device like /dev/null.
- Do you think it would be worthwhile making it warn instead of crash?  If there were warnings then do you think that would have been noticed when you hit the problem motivating this commit?